include/rmm/cuda_stream_pool.hpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <rmm/cuda_stream.hpp>
+#include <rmm/cuda_stream_view.hpp>
+
+#include <atomic>
+#include <vector>
+
+namespace rmm {
+
+/**
+ * @brief A pool of CUDA streams.
+ *
+ * Provides fast access to a circular buffer of `cuda_stream` objects.
+ *
+ */
+class cuda_stream_pool {
+ public:
+  static constexpr std::size_t default_size{16};  ///< Default stream pool size
```

Comment:
Should this be private?

Reply:
It's a default parameter to a public constructor, so I assumed it must be public...

Reply:
The constructor has access to private fields. There is no reason for any caller to change this default, right?

Reply:
From what I remember, Cython doesn't really work well with default args, so they usually need access to the definition of what is being used as the default.

Reply:
This is constexpr, so a caller cannot change it. I made it public because I thought it might be useful for users to be able to access the default size at run time.